pkg/code-generator/codegen/templates/snapshot_emitter_template.go

```diff
@@ -93,12 +93,16 @@ func init() {
 	)
 }
 
+type {{ .GoName }}SnapshotEmitter interface {
+	Snapshots(watchNamespaces []string, opts clients.WatchOpts) (<-chan *{{ .GoName }}Snapshot, <-chan error, error)
+}
+
 type {{ .GoName }}Emitter interface {
+	{{ .GoName }}SnapshotEmitter
```

Comment:
the naming here is confusing. what's the difference between an emitter and a snapshotemitter?

Reply:
the event loop only needs the Snapshots() function hence the separation of interfaces.
This make the eventloop easier to test. Plus since it doesn't need to full interface, no need to expose it.

Reply:
ok but `Emitter` and `SnapshotEmitter` are interchangeable in terms of their meaning. can we come up with names that make it clear?

also, i'm not sure it's actually important for us to create a new interface type. since Emitter (as far as i know) is always generated

Reply:
i want to use the new interface as the event loop argument as the event loop doesn't need the other methods.
what names do you have in mind?